### PR TITLE
All Unique Permutations in C++

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This is a google interview question.
Code has been added in cp(competitive programming) folder.
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5396

### CODE REVIEW

**Comment by gayatri517** [comment]
@ankitaggarwal23 Kindly review my PR

File: C-Plus-Plus/cp/all_unique_permutation.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*Problem Statement:
+Given a collection of numbers that might contain duplicates, 
+return all possible unique permutations excluding the user input. */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/all_unique_permutation.cpp]: Describe it properly

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
resolve merge conflicts.

**Comment by gayatri517** [comment]
@HarshCasper @ankitaggarwal23 I have made the changes and resolved the merge conflict
Kindly review it

**Comment by rohanbabbar04** [comment]
@gayatri517 
You still have merge conflicts

**Comment by gayatri517** [comment]
@HarshCasper @ankitaggarwal23 i have resolved the merge conflicts. Please review now

File: C-Plus-Plus/cp/all_unique_permutation.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    for(int j=i;in[j]!='\0';j++)
+    {
+       swap(in[i],in[j]);
+       permute(in,i+1);
+       /*Backtracking so that more permutations could
+       be generated*/
+       swap(in[i],in[j]);
+    }
+
```
Comment (bhav09) [inline C-Plus-Plus/cp/all_unique_permutation.cpp:23]: Remove extra lines of code. These new lines are redundant to the code file.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Minor Changes. Reviewed.

**Comment by gayatri517** [comment]
@bhav09 I have made the changes and resolved the conflict also. Please look at it.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**